public/components/trace_analytics/components/services/service_view.tsx

```diff
@@ -118,16 +114,31 @@ export function ServiceView(props: ServiceViewProps) {
           },
           {
             text: props.serviceName,
-            href: `#/services/${encodeURIComponent(props.serviceName)}`,
+            href: (() => {
+              const dataSourceId = props.dataSourceMDSId[0].id;
+              if (dataSourceId && dataSourceId !== '') {
+                return `#/services?datasourceId=${encodeURIComponent(
+                  dataSourceId
+                )}&serviceId=${encodeURIComponent(props.serviceName)}`;
+              } else {
+                return `#/services?serviceId=${encodeURIComponent(props.serviceName)}`;
+              }
+            })(),
           },
         ]
       );
-  }, [props.serviceName]);
-
-  const DataSourceMenu = props.dataSourceManagement?.ui?.getDataSourceMenu<DataSourceViewConfig>();
+    props.setDataSourceMenuSelectable?.(false);
+  }, [props.serviceName, props.setDataSourceMenuSelectable]);
 
   const redirectToServicePage = (service: string) => {
-    window.location.href = `#/services/${service}`;
+    const dataSourceId = props.dataSourceMDSId[0].id;
+    if (dataSourceId && dataSourceId !== '') {
+      window.location.href = `#/services?datasourceId=${encodeURIComponent(
+        dataSourceId
+      )}&serviceId=${encodeURIComponent(service)}`;
+    } else {
+      window.location.href = `#/services?datasourceId=&serviceId=${encodeURIComponent(service)}`;
+    }
```

Comment:
if we are duplicating code might as well use a function for this.

Reply:
export const generateServiceUrl = (service: string, dataSourceId: string) => {
  const url = `#/services?serviceId=${encodeURIComponent(service)}`;

  if (dataSourceId && dataSourceId !== '') {
    return `${url}&datasourceId=${encodeURIComponent(dataSourceId)}`;
  }

  return `${url}&datasourceId=`;
};